tools/alara_output_processing/alara_output_plotting.py

```diff
@@ -209,6 +236,209 @@ def construct_legend(ax, data_comp=False):
+            sub-list for each row in the table, corresponding to the total
+            number of nuclides represented within the table (and in all of the
+            pie charts in the grid). Each row-list contains a floating point
+            number representing the percentage contribution of that nuclide at
+            each cooling time. 
+    '''
+
+    table_data = []
+    for nuc in all_nucs:
+        row = []
+        for t in times:
+            ts = agg[(agg['nuclide'] == nuc) & (agg['time'] == t)]
+            pct = float(ts['value'].iloc[0])*100 if not ts.empty else 0.0
+            row.append(f'{pct:.1f}%')
+        table_data.append(row)
+
+    return table_data
+
+def sort_by_index(lst, indices):
+    '''
+    Reorganize a list by a new list of indices.
+
+    Arguments:
+        lst (list): List to be reordered.
+        indices (list of int): Indices for reordering.
+
+    Returns:
+        reordered_lst (lst): Modified copy of lst, reordered to the new
+            indices. 
+    '''
+
+    return [lst[i] for i in indices]
+
+def pie_grid_relative_tables(
+    fig, agg, all_nucs, times, nrows, ncols, time_unit, color_map
+):
+    '''
+    Given a Matplotlib Figure object, create a subplot Matplotlib table object
+        to act as a color-coordinated legend for a grid of pie charts with
+        the percentage contributions of each nuclide represented in any of the
+        pie charts at each cooling time (as well as a row for aggregated
+        "Other" nuclides taht fall below a given proportional threshold).
+
+    Arguments:
+        fig (matplotlib.figure.Figure): Matplotlib Figure object wherein the
+            grid of pie charts will be constructed.
+        agg (alara_output_processing.ALARADFrame): Processed ALARADFrame
+            (potentially) with new "Other" rows for each time, containing all
+            aggregated data below the thresholds.
+        all_nucs (list of str): List of all nuclide names included in any of
+            the pie charts in the grid.
+        times (list of floats): Chronological list of cooling times.
+        nrows (int): Number of rows in the grid of pie charts.
+        ncols (int): Number of columns in the grid of pie charts.
+        time_unit (str): Units for cooling times.
+        color_map (dict): Pre-constructed color map for each nuclide to match
+            table rows with pie chart wedges.
+
+    Returns:
+        gs (matplotlib.gridspec.GridSpec): Matplotlib Gridspec object to
+            organize the combined set of subplots containing the resultant
+            table as well as the individual pie charts, to be added
+            subsequent to this function's calling.
+    '''
+
+    # Create table gridspec
+    left = 0.05
+    top = 0.95
+    gs = fig.add_gridspec(
+        nrows + 1, ncols,
+        height_ratios=([1.0] + [1]*nrows),
+        hspace=0.4,
+        wspace=0.3,
+        left=left, right=(1-left), top=top, bottom=(1-top)
+    )
```

Comment:
Separation of concerns... I think the `GridSpec` should be generated at a different level, and then the specific `table_ax` passed into this method to populate it.